src/indexer/segment_updater.rs

```diff
@@ -629,6 +629,36 @@ impl SegmentUpdater {
         scheduled_result
     }
 
+    pub(crate) fn merge_foreground(
+        &self,
+        merge_operation: MergeOperation,
+    ) -> crate::Result<Option<SegmentMeta>> {
+        assert!(
+            !merge_operation.segment_ids().is_empty(),
+            "Segment_ids cannot be empty."
+        );
```

Comment:
Can this be validated in the factory function?

Reply:
This code is a cut-n-paste from the other version of merging that happens in the background.  So my preference with that was to keep the code identical (where possible).

I don't have the time at moment to do some better redesign of tantivy's merging interfaces such that the existing background path can share as much code as possible with the new foreground path.  :(